### ISSUE TITLE
Rerun failed tests broken in 5.4.3

### ISSUE THREAD

**Issue by throwable-one** [pytest-dev/pytest#7508]
```
from unittest import TestCase


def test_test():
    assert False


class FooTest(TestCase):
    def test_test(self):
        self.fail()
```

Run pytest. Two tests should fail. Run with `` --last-failed``. Only one test launched.
Try latest 5.3: it works as expected

**Comment by The-Compiler** [comment]
Bisected to 1b30514783d8ed4225a59af3f9f3fa668ab56f67 ("LFPlugin: use sub-plugins to deselect during collection (#6448)") which is a fix for #5301 - cc @blueyed

**Comment by blueyed** [comment]
Likely fixed by https://github.com/pytest-dev/pytest/pull/6991, was not backported (https://github.com/pytest-dev/pytest/pull/7219).

**Comment by The-Compiler** [comment]
Indeed fixed via #6991 in master already - sorry, I missed that! Closing this then.
